app/src/main/java/com/github/swent/echo/compose/components/HomeScreen.kt

```diff
@@ -8,118 +8,105 @@ import androidx.compose.material3.ModalNavigationDrawer
-            from = remember { mutableStateOf(ZonedDateTime.now()) },
-            to = remember { mutableStateOf(ZonedDateTime.now()) },
-            sortBy = remember { mutableStateOf(SortBy.NONE) }
-        )
+
+    val mode by homeScreenViewModel.mode.collectAsState()
+    val overlay by homeScreenViewModel.overlay.collectAsState()
+
+    val displayEventInfo by homeScreenViewModel.displayEventInfo.collectAsState()
+    val displayEventList by homeScreenViewModel.displayEventList.collectAsState()
+
+    val filters by homeScreenViewModel.filtersContainer.collectAsState()
 
     Box(modifier = Modifier.padding(paddingValues)) {
-        if (mode.value == MapOrListMode.LIST) {
-            ListDrawer(SAMPLE_EVENTS)
+        if (mode == MapOrListMode.LIST) {
+            ListDrawer(displayEventList)
         } else {
             MapDrawer(
-                events = SAMPLE_EVENTS,
-                callback = { event -> onEventSelected(event) },
+                events = displayEventList,
+                callback = { event -> homeScreenViewModel.onEventSelected(event) },
             )
         }
         // add the tag filtering here
-        if (overlay.value == Overlay.EVENT_INFO_SHEET && displayEventInfo.value != null) {
+        if (overlay == Overlay.EVENT_INFO_SHEET && displayEventInfo != null) {
             EventInfoSheet(
-                event = displayEventInfo.value!!,
+                event = displayEventInfo!!,
                 onJoinButtonPressed = {},
                 onShowPeopleButtonPressed = {},
-                onDismiss = { overlay.value = Overlay.NONE },
+                onDismiss = { homeScreenViewModel.clearOverlay() },
                 onFullyExtended = {}
             )
             // {navActions.navigateTo(Routes.EventInfoScreen)}) <- when we make a whole screen for
             // the event info
         }
-        if (overlay.value == Overlay.SEARCH_SHEET) {
+
+        if (overlay == Overlay.SEARCH_SHEET) {
             SearchMenuSheet(
                 filters,
                 onFullyExtended = {},
-                onDismiss = { overlay.value = Overlay.NONE }
+                onDismiss = { homeScreenViewModel.clearOverlay() },
+                searchEntryCallback = { s -> homeScreenViewModel.onSearchEntryChanged(s) },
+                epflCallback = { homeScreenViewModel.onEpflCheckedSwitch() },
+                sectionCallback = { homeScreenViewModel.onSectionCheckedSwitch() },
+                classCallback = { homeScreenViewModel.onClassCheckedSwitch() },
+                pendingCallback = { homeScreenViewModel.onPendingCheckedSwitch() },
+                confirmedCallback = { homeScreenViewModel.onConfirmedCheckedSwitch() },
+                fullCallback = { homeScreenViewModel.onFullCheckedSwitch() },
+                sortByCallback = { sortBy -> homeScreenViewModel.onSortByChanged(sortBy) },
+                resetFiltersCallback = { homeScreenViewModel.resetFiltersContainer() }
```

Comment:
Here you could use a simper syntax, e.g. `onDismiss = homeScreenViewModel::clearOverlay`, avoids wrapping functions into functions. This also works for functions with arguments as long as the signature is the same, e.g. `searchEntryCallback = homeScreenViewModel::onSearchEntryChanged`. Note that there are a lot of other places you could use this syntax, but I didn't add a comment every time.

Reply:
oh that's good I'll use that